### PR TITLE
Add module to load and generate balx files for cli commands at build time

### PR DESCRIPTION
**Pull request by NatashaWso2**
- ballerina pull
- ballerina push
- ballerina search

### CODE REVIEW

File: cli/ballerina-cliutils/pom.xml [hunk 1]
```diff
@@ -0,0 +1,186 @@
+<!--
```
Comment (manuranga) [inline cli/ballerina-cliutils/pom.xml]: shall we rename cliutils to cli-utils everywhere.

File: cli/ballerina-cliutils/src/main/ballerina/pull/ballerina.pull.bal [hunk 2]
```diff
@@ -1,3 +1,5 @@
+package ballerina.pull;
```
Comment (manuranga) [inline cli/ballerina-cliutils/src/main/ballerina/pull/ballerina.pull.bal]: let's not use ballerina as a pkg name.

File: compiler/ballerina-lang/src/main/java/org/ballerinalang/util/EmbeddedExecutorProvider.java [hunk 3]
```diff
@@ -10,6 +10,11 @@
  */
 public class EmbeddedExecutorProvider {
     private static EmbeddedExecutorProvider provider;
+
+    static {
+        provider = new EmbeddedExecutorProvider();
```
Comment (manuranga) [inline compiler/ballerina-lang/src/main/java/org/ballerinalang/util/EmbeddedExecutorProvider.java]: why static block, cant we do a single line in line 12.

File: cli/ballerina-cliutils/pom.xml (hunk 1 above)
Comment (NatashaWso2) [reply cli/ballerina-cliutils/pom.xml]: Fixed with 9fcdece5175329f9bb65d427cb38db2c9f44d701

File: cli/ballerina-cliutils/src/main/ballerina/pull/ballerina.pull.bal (hunk 2 above)
Comment (NatashaWso2) [reply cli/ballerina-cliutils/src/main/ballerina/pull/ballerina.pull.bal]: Fixed with 9fcdece5175329f9bb65d427cb38db2c9f44d701

File: compiler/ballerina-lang/src/main/java/org/ballerinalang/util/EmbeddedExecutorProvider.java (hunk 3 above)
Comment (NatashaWso2) [reply compiler/ballerina-lang/src/main/java/org/ballerinalang/util/EmbeddedExecutorProvider.java]: Fixed with 9fcdece5175329f9bb65d427cb38db2c9f44d701

**Pull request merged**